maven35-agent/pom.xml

```diff
@@ -24,12 +24,48 @@
     <dependency>
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-classworlds</artifactId>
+      <version>2.5.2</version>
     </dependency>
     <dependency>
       <groupId>org.eclipse.sisu</groupId>
       <artifactId>org.eclipse.sisu.plexus</artifactId>
-      <version>0.0.0.M2a</version>
+      <version>0.3.3</version>
     </dependency>
+    
+   <!-- Upper bounds from org.apache.maven:maven-embedder:3.5.0 -->
```

Comment:
I think this may be the wrong approach. See my comments in [JENKINS-45271](https://issues.jenkins-ci.org/browse/JENKINS-45271).